src/libexpr/primops.cc

```diff
@@ -2485,15 +2484,7 @@ static void addPath(
             // FIXME: handle CA derivation outputs (where path needs to
             // be rewritten to the actual output).
             auto rewrites = state.realiseContext(context);
-            path = {state.rootFS, CanonPath(state.toRealPath(rewriteStrings(path.path.abs(), rewrites), context))};
-
-            try {
-                auto [storePath, subPath] = state.store->toStorePath(path.path.abs());
-                // FIXME: we should scanForReferences on the path before adding it
-                refs = state.store->queryPathInfo(storePath)->references;
-                path = {state.rootFS, CanonPath(state.store->toRealPath(storePath) + subPath)};
-            } catch (Error &) { // FIXME: should be InvalidPathError
-            }
```

Comment:
Is the idea that `refs` is unused? If so, let's also delete the variable above? If not... what's going on?

Reply:
This is a regression in....ea38605d116e1c8db55a8a8cb7400724b931458d?

Reply:
Yeah this appears to be dead code. I think the idea was to propagate the references, but that "feature" was never really well thought out. Maybe we should warn/fail if the path has references...

Reply:
We can investigate what happened here later. IIRC there was some regressions with this too.